server/routes/nl.py

```diff
@@ -294,34 +317,56 @@ def _detection(orig_query, cleaned_query) -> Detection:
   main_place = None
   resolved_places = []
 
+  # Start updating the query_detection_debug_logs. Create space for place dcid inference
+  # and place resolution. If they remain empty, the function belows were never triggered.
+  query_detection_debug_logs["place_dcid_inference"] = {}
+  query_detection_debug_logs["place_resolution"] = {}
   # Look to find place DCIDs.
   if places_str_found:
-    place_dcids = _infer_place_dcids(places_str_found)
+    place_dcids = _infer_place_dcids(
+        places_str_found, query_detection_debug_logs["place_dcid_inference"])
     logging.info(f"Found {len(place_dcids)} place dcids: {place_dcids}.")
 
-    # Step 2: replace the places in the query sentence with "".
```

Comment:
@pradh this is the fix for the bug we discussed. This needed to go under the following conditional (if place_dcids:).

Reply:
Could you elaborate more on what the problem was?  And maybe add some comments on why this should go under resolved_places, rather than say `if place_dcids`?

Reply:
I am not sure if it should be under `if place_dcids` or under `if resolved_places` because I have not seen resolution failures very often, if at all? But I have put this back under `if place_dcids` to have parity with the previous behavior and also added a detailed comment with a possible TODO (to investigate if only those place strings should be removed which get a DCID--this would be another change from existing/previous behavior so should first be tested).